src/main/scala/tilelink/Parameters.scala

```diff
@@ -1087,6 +1199,64 @@ case class TLEdgeParameters(
 
   val bundle = TLBundleParameters(client, manager)
   def formatEdge = client.infoString + "\n" + manager.infoString
+
+  private def emitHelper(
+    safe:    Boolean,
+    member: TLMasterParameters => TransferSizes,
+    address: UInt,
+    lgSize:  UInt,
+    range:   Option[TransferSizes]): Bool = {
+    def trim(x: TransferSizes) = range.map(_.intersect(x)).getOrElse(x)
+    // groupBy returns an unordered map, convert back to Seq and sort the result for determinism
+    val supportCases = groupByIntoSeq(client.clients)(m => trim(member(m))).map { case (k, vs) =>
+      k -> vs.flatMap(_.visibility)
+    }
+    val mask = if (safe) ~BigInt(0) else AddressDecoder(supportCases.map(_._2))
+    val simplified = supportCases.map { case (k, seq) => k -> AddressSet.unify(seq.map(_.widen(~mask)).distinct) }
+    simplified.map { case (s, a) =>
+      (Bool(Some(s) == range) || s.containsLg(lgSize)) &&
+      a.map(_.contains(address)).reduce(_||_)
+    }.foldLeft(Bool(false))(_||_)
+  }
+
+  private def supportHelper(
+    safe:    Boolean,
+    member: TLSlaveParameters => TransferSizes,
+    address: UInt,
+    lgSize:  UInt,
+    range:   Option[TransferSizes]): Bool = {
+    def trim(x: TransferSizes) = range.map(_.intersect(x)).getOrElse(x)
+    // groupBy returns an unordered map, convert back to Seq and sort the result for determinism
+    val supportCases = groupByIntoSeq(manager.managers)(m => trim(member(m))).map { case (k, vs) =>
+      k -> vs.flatMap(_.address)
+    }
+    val mask = if (safe) ~BigInt(0) else AddressDecoder(supportCases.map(_._2))
+    val simplified = supportCases.map { case (k, seq) => k -> AddressSet.unify(seq.map(_.widen(~mask)).distinct) }
+    simplified.map { case (s, a) =>
+      (Bool(Some(s) == range) || s.containsLg(lgSize)) &&
+      a.map(_.contains(address)).reduce(_||_)
+    }.foldLeft(Bool(false))(_||_)
+  }
```

Comment:
This function already exists on TLSlavePortParameters. Why is it being duplicated here?